docs/model-deployment-prepackaged-server-custom.md

```diff
@@ -13,15 +13,159 @@ description: Custom Pre-packaged Server
+2. check if there is a class named Model in the loaded module.
+3. create a Model object
+
+The concepts of the pseudo code will look like this:
+
+```
+# create a user_model and delegate client calls to it
+user_model = Model(**parameters)
+
+# load model if load method implemented
+user_model.load()
+
+# response the result of the predict method
+user_model.predict(features, feature_names, **kwargs)
+```
+
+The model is a simple Python class, and you have to implement the `predict` to send the result of predictions:
+
+```python
+class Model:
+    def __init__(self, model_uri=None):
+        # initialization
+        # 1. configure model path from the model_uri if needed
+        self.model_uri = model_uri
+        self.model = None
+
+        # 2. initialize the predictor
+        # you might want to enable GPU if it is not enabled automatically
+
+        # 3. invoke load method to preload the model
+        self.ready = False
+        self.load()
+
+    def load(self):
+        # load and create the model
+        # if model_uri was given, load data and create model instance from it
+        if self.ready:
+            return
+
+        # build model
+        # 1. set to self.model
+        # 2. make set.ready = True
+        self.ready = True
+
+    def predict(self, X, feature_names = None, meta = None):
+        # execute self.model.predict(X)
+        print(X, feature_names, meta)
+        return "Hello Model"
+```
+
+
+## Handle model files
+
+The model files could ship with your image, you could make it more feasible with the `model_uri` mechanism of the PrimeHub Deployment.
+
+PrimeHub users could create a Deployment with Model URI:
+![](assets/mdeploy_quickstart_deploydetail_1_phfs.png)
+
+In the container's view, the model files will mount to a local filesystem, such as `/mnt/models`:
+
+```python
+def __init__(self, model_uri=None):
+    self.model_uri = model_uri
+    ...
+```
+
+Please add the `model_uri` argument to the `__init__` function and make sure having `None` as the default value. It means your Model supporting with or without `model_uri`. If a user give a `Model URI` value, the `__init__` will get a mount path from the variable `model_uri`. You can check which files should be loaded in that path.
+
+It is very common to write a load method to load and build model instance:
+
+```python
+  def load(self):
+    ...
+```
+
+You could build the model instance in the `__init__`, if your loading process is very simple. The `load` method is optional.
+
+## MLflow model
+
+PrimeHub Deployment could work with MLflow Model Registry. When a user set `Model URI` starting with `models:`, it will get model files from MLflow Model Registry. Nothing special with the model files comes from the MLflow Model Registry, it also mounts to the `model_uri`. However, there is a convention to let you know:
+
+> The model files always ships with the `MLmodel` file.
+
+You should register a model with the `MLmodel` file
```

Comment:
PrimeHub Deployment could work with MLflow Model Registry. When a user set `Model URI` starting with `models:`, it will get model files from MLflow Model Registry. From the perspective of model server, there is nothing special that the model files downloaed and copied to the `/mnt/model_uri` in the local filesystem.

Reply:
About the following paragraph, it is actually not true. It is convention for us, not for custom pre-pacakged server
"""
However, there is a convention to let you know:

> The model files always ships with the `MLmodel` file.
You should register a model with the `MLmodel` file
"""

We can skip this all paragraph.